webapp/src/utils/api.ts

```diff
@@ -88,7 +89,18 @@ export const fetchApi =
     // fetch() might throw if the user is offline, or some unlikely networking error occurs, such a DNS lookup failure.
     // Let's also throw if the status is not OK, so it's uniform.
     if (!response.ok) {
-      throw Error(`${response.status} : ${response.statusText}`);
+      const httpExceptionModelCodes: number[] = [
+        400, 401, 403, 404, 422, 500, 503,
+      ];
+      if (httpExceptionModelCodes.includes(response.status)) {
+        const { detail: errorMessage } =
+          (await response.json()) as HTTPExceptionModel;
+        throw Error(errorMessage);
+      } else {
```

Comment:
Instead of assuming which status code comes with JSON or not, have you considered simply `try`ing to call `.json()` and default to the generic message if you `catch` and exception?
```suggestion
      try {
        const { detail: errorMessage } =
          (await response.json()) as HTTPExceptionModel;
        throw Error(errorMessage);
      } catch {
```

Reply:
Oups! There was an issue with my suggestion. I'm sorry! It should have been tested before committing it. We `throw` inside the `try`, so it will always `catch` our error. 😅 
We can revert to your initial implementation with `httpExceptionModelCodes`. This only thing I would change would be to make `httpExceptionModelCodes` a constant at the top of the file. I suggest `HTTP_EXCEPTION_STATUS_CODES`.